google/cloud/firestore_v1/aggregation.py

```diff
@@ -154,15 +193,29 @@ def _make_stream(
 
             if response is None:  # EOI
                 break
+
+            if metrics is None and response.explain_metrics:
+                metrics = response.explain_metrics
+
             result = _query_response_to_result(response)
-            yield result
+            if len(result) > 0 or metrics is None:
+                # When explain_options.analyze == False, the query isn't run
+                # but query profiling results are returned. In this case we
+                # do not yield an empty result.
+                yield result
+
+        return metrics
```

Comment:
I'm realizing AsyncGenerator doesn't support returning values, so we'll have to find a different method for the async side... I don't know if that changes anything here

Reply:
I encountered this issue when implementing the async part. The temp hacky solution is to yield the metrics and catch it by checking type 😂 I think I really like the return here so I'd rather the sync and async aren't symmetric here, but let's discuss in more details in our meeting.

Reply:
Yeah, I guess we're back to our old worse options for the Async side! I agree it's probably fine to keep them separate if it saves us headaches on the sync side at least